src/routes/api/v1/search-index/reset/+server.ts

```diff
@@ -0,0 +1,50 @@
+import { json } from "@sveltejs/kit";
+import { loadPageMetadataMap } from "$lib/loadPageMetadataMap";
+import algoliasearch from "algoliasearch";
+import { PUBLIC_ALGOLIA_APP_ID, PUBLIC_ALGOLIA_INDEX } from "$env/static/public";
+import { ALGOLIA_API_KEY } from "$env/static/private";
+const algoliaClient = algoliasearch(PUBLIC_ALGOLIA_APP_ID, ALGOLIA_API_KEY);
+
+/**
+ * This endpoint does a wholesale reset of the page metadata map in Algolia.
+ * Since it requires each record to have an objectID, a reset call won't result
+ * in any duplication of metadata records in the Algolia index, just a replacement
+ * of the previous object value with the current value in contentful
+ * TODO: secure this somehow to prevent unnecessary operations
+ **/
```

Comment:
This isn't a security concern in the sense that it would wipe out the index, but for limiting overall operations in Algolia.

As it currently works, it just grabs the `pageMetdataMap` and updates all of those records in Algolia with the current values. Adding some kind of credentialed check in front of this call would be mainly to prevent anyone from racking up update calls, [which has monthly limits](https://support.algolia.com/hc/en-us/articles/4406981829777-How-does-Algolia-count-records-and-operations-).